parlai/scripts/self_chat.py

```diff
@@ -86,15 +98,31 @@ def _run_self_chat_episode(opt, world, world_logger):
 
 def self_chat(opt):
     random.seed(opt['seed'])
+    partner = opt['partner_model_file']
+    partner_opt_file = opt['partner_opt_file']
 
     # Create agents
     agent1 = create_agent(opt, requireModelExists=True)
-    agent2 = agent1.clone()
+    if not partner:
+        # Self chat with same model
+        agent2 = agent1.clone()
+    else:
+        # Self chat with different models
+        if partner_opt_file:
+            print(f"WARNING: Loading model with opts from {partner_opt_file}")
+            with open(partner_opt_file) as f:
+                partner_opt = json.load(f)
+            for k, v in partner_opt.items():  # override partner opts
```

Comment:
let's add a comment here explaining what we're overrriding: like `if k in opt and opt[k] != v` add a print "Warning: overriding option [ {k}: {old} => {v}]'."